### PR TITLE
Add HorribleSubs torrent provider

### PR DESCRIPTION
**Pull request by medariox**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: medusa/providers/torrent/html/horriblesubs.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+                    leechers = 0
+                    size = -1
+
+                    item = {
+                        'title': title,
+                        'link': download_url,
+                        'size': size,
+                        'seeders': seeders,
+                        'leechers': leechers,
+                        'pubdate': None,
```
Comment (duramato) [inline medusa/providers/torrent/html/horriblesubs.py]: HorribleSubs doesn't provide publication date? Maybe add a comment with that too

Comment (medariox) [inline medusa/providers/torrent/html/horriblesubs.py]: Most providers don't support pubdate and we don't have any comment for those either, wanted to keep it consistent that way.

Comment (duramato) [inline medusa/providers/torrent/html/horriblesubs.py]: Not having it implemented it's different than not providing the info at all.

File: medusa/providers/torrent/html/horriblesubs.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+"""Provider code for HorribleSubs."""
+
+from __future__ import unicode_literals
+
+import traceback
+
+from dateutil import parser
+
+from medusa import (
+    logger,
```
Comment (labrys) [inline medusa/providers/torrent/html/horriblesubs.py]: Use new style logging:
```
import logging
from medusa.logger.style.adapter import BraceAdapter

log = BraceAdapter(logging.getLogger(__name__))
log.logger.addHandler(logging.NullHandler())

log.debug('Brace formated {msg}', {'msg': 'is preferred'})
```

Comment (medariox) [inline medusa/providers/torrent/html/horriblesubs.py]: Done

**Pull request merged**